### ISSUE TITLE
Update needed for Python 3.8

### ISSUE THREAD

**Issue by RobertoPrevato** [Kludex/uvicorn#417]
Hi,
A quick information, maybe will save a bit of time for someone.

1. The current version of `uvloop` (0.13.0) does not work at runtime with Python 3.8, it cannot be imported because it uses a method that was removed: 
```
  File "uvloop/includes/stdlib.pxi", line 114, in init uvloop.loop
AttributeError: module 'sys' has no attribute 'set_coroutine_wrapper'
```
2. This was fixed in the last version of uvloop from GitHub (uvloop 0.14.0.dev0), so compiling from source is fine
3. However, it cannot be installed with uvicorn because of the requirement in its setup.py
```
ERROR: uvicorn 0.8.6 has requirement uvloop==0.12.*, but you'll have uvloop 0.14.0.dev0 which is incompatible.
```

In conclusion, to support Python 3.8 and install from PyPi, we need to wait for a new version of uvloop and a new version of uvicorn with updated dependency version.

**Comment by lovelydinosaur** [comment]
Okay, thanks - that's really helpful.

I think the route that we *may* eventually want to go down is making the *default* install of uvicorn be the most basic possible, and only install stuff like `uvloop` and `httptools` with `pip install uvicorn[accelerated]` or something like that.

**Comment by lovelydinosaur** [comment]
Also I think it'd probably okay for us to pin `uvloop` in a more relaxed way, and instead have it as `uvloop==0.*` - That would at least help mitigate this issue.

**Comment by lovelydinosaur** [comment]
Alrighty - our latest release is pinned to `uvloop==0.*`.
We probably want to add 3.8 to our test matrix, with 0.14.0.dev0 installed.

**Comment by stonebig** [comment]
you can also specify `uvloop~=0.12.0`  to mean anything >=0.12.0 <1.0

**Comment by fenchu** [comment]
3.8.0 has been out for 4 days. please release 0.14, I have a responder project I'd love to use walrus := in :-)

**Comment by gvbgduh** [comment]
It's getting a bit complicated, there's no published or tagged version which is `0.14.0dev`. It sits in the master branch dated by Apr, 23 and the actual fix was a bit later. So, it takes an install from a certain commit that also requires bringing the `Cython` lib to the scope.
I have a passing configuration in https://github.com/encode/uvicorn/pull/453.
I might be missing a simpler way to approach it, but this one doesn't look very nice. And I'm still having some difficulties to build it on macOS.

**Comment by lovelydinosaur** [comment]
Closing this issue - we're waiting on an update to uvloop - nothing else for us to do here on this project.

**Comment by iwoloschin** [comment]
@tomchristie, is it appropriate to re-use this issue to request a `uvloop` bump to 0.14 now that it's released?

https://github.com/MagicStack/uvloop/releases/tag/v0.14.0

**Comment by lovelydinosaur** [comment]
Absolutely, yup! https://github.com/encode/uvicorn/pull/485
